log.go

```
@@ -100,12 +114,73 @@ type Logger struct {
 	f, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0644)
 	if err != nil {
-		return Logger{}, err
+		return &Logger{}, err
 	}
 	return New(level, f, sentry, sentryTags)
+
+}
+
+// LogToDir creates a new Logger that writes to a file in the directory Path. The file will be named
+// path/fileNameBase.YYYY.MM.DD:HH:mm:SS.log. If the file doesn't exist, it
+// will be created. If it does exist, new log lines will be appended to it.
+// Loggers will be rotatable iif they are created with this function.
+//
+// If sentry is non-empty, it will be used as a DSN to connect to a Sentry error collector. The sentryTags
+// are a key/value mapping that will be applied to your Sentry errors. You can use them to set things like
+// the version of your software running, etc.
+func LogToDir(level Level, path string, fileNameBase string, sentry string, sentryTags map[string]string) (*Logger, error) {
+	fullPath := calculateFileName(path, fileNameBase)
+	f, err := os.OpenFile(fullPath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0644)
+	if err != nil {
+		return &Logger{}, err
+	}
+	logger, err1 := New(level, f, sentry, sentryTags)
+	logger.logFile = f
+	logger.logFileBase = fileNameBase
+	logger.logFileDir = path
+
+	rotatableLoggers.PushBack(logger)
+	return logger, err1
+}
+
+// RotateLogFile creates a new logfile based on the Logger's filepath, base file name, and
+// the current date/time. This function will only work with Loggers that have been created
+// using LogToDir. This function will silently fail if the Logger was not created using
+// LogToDir, and is missing a base filename or a directory.
+//
+// When SIGUSR1 is caught, RotateLogFile will be automatically called on all Loggers created
+// using LogToDir.
+func (l *Logger) RotateLogFile() {
+	if l.logFile != nil && l.logFileBase != "" && l.logFileDir != "" {
+		newPath := calculateFileName(l.logFileDir, l.logFileBase)
+		newFile, _ := os.OpenFile(newPath, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0644)
```

Ignoring errors gives me the heebie jeebies.